src/aggregate/aggregate_exec.c

```diff
@@ -1137,82 +1139,88 @@ typedef struct {
-      }
-
-      if (RedisModule_StringToLongLong(argv[5], &count) != REDISMODULE_OK) {
-        RedisModule_ReplyWithErrorFormat(ctx, "Bad value for COUNT: `%s`", RedisModule_StringPtrLen(argv[5], NULL));
-        RedisModule_EndReply(reply);
-        return REDISMODULE_OK;
-      }
+  long long count = 0;
+  if (argc > 4) {
+    // e.g. 'COUNT <chunk size>'
+    // Verify that the 4'th argument is `COUNT`.
+    if (strcasecmp(RedisModule_StringPtrLen(argv[4], NULL), "count") != 0) {
+      return RedisModule_ReplyWithError(ctx, "Unknown argument at offset 4");
     }
-    // We have to check that we are not blocked yet from elsewhere (e.g. coordinator)
-    if (RunInThread() && !RedisModule_GetBlockedClientHandle(ctx)) {
-      CursorReadCtx *cr_ctx = rm_new(CursorReadCtx);
-      cr_ctx->bc = RedisModule_BlockClient(ctx, NULL, NULL, NULL, 0);
-      cr_ctx->cid = cid;
-      cr_ctx->count = count;
-      RedisModule_BlockedClientMeasureTimeStart(cr_ctx->bc);
-      workersThreadPool_AddWork((redisearch_thpool_proc)cursorRead_ctx, cr_ctx);
-    } else {
-      cursorRead(reply, cid, count, false);
+
+    if (argc < 6) {
+      return RedisModule_ReplyWithError(ctx, "COUNT requires an argument");
     }
-  } else if (strcasecmp(cmd, "DEL") == 0) {
-    int rc = Cursors_Purge(GetGlobalCursor(cid), cid);
-    if (rc != REDISMODULE_OK) {
-      RedisModule_Reply_Error(reply, "Cursor does not exist");
-    } else {
-      RedisModule_Reply_SimpleString(reply, "OK");
+
+    if (RedisModule_StringToLongLong(argv[5], &count) != REDISMODULE_OK) {
+      return RedisModule_ReplyWithError(ctx, "Bad value for COUNT");
     }
-  } else if (strcasecmp(cmd, "GC") == 0) {
-    int rc = Cursors_CollectIdle(&g_CursorsList);
-    rc += Cursors_CollectIdle(&g_CursorsListCoord);
-    RedisModule_Reply_LongLong(reply, rc);
+  }
+
+  // We have to check that we are not blocked yet from elsewhere (e.g. coordinator)
+  if (RunInThread() && !RedisModule_GetBlockedClientHandle(ctx)) {
+    CursorReadCtx *cr_ctx = rm_new(CursorReadCtx);
+    cr_ctx->bc = RedisModule_BlockClient(ctx, NULL, NULL, NULL, 0);
+    cr_ctx->cid = cid;
+    cr_ctx->count = count;
+    RedisModule_BlockedClientMeasureTimeStart(cr_ctx->bc);
+    workersThreadPool_AddWork((redisearch_thpool_proc)cursorRead_ctx, cr_ctx);
   } else {
-    RedisModule_Reply_Error(reply, "Unknown subcommand");
+    cursorRead(ctx, cid, count, false);
   }
-  RedisModule_EndReply(reply);
+
   return REDISMODULE_OK;
 }
+
+/**
+ * FT.CURSOR DEL {index} {CID}
+ */
+int RSCursorDelCommand(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
+  if (argc != 4) {
+    return RedisModule_WrongArity(ctx);
+  }
+
+  long long cid;
+  if (RedisModule_StringToLongLong(argv[3], &cid) != REDISMODULE_OK) {
+    return RedisModule_ReplyWithError(ctx, "Bad cursor ID");
+  }
+
+  if (Cursors_Purge(GetGlobalCursor(cid), cid) != REDISMODULE_OK) {
+    return RedisModule_ReplyWithError(ctx, "Cursor does not exist");
```

Comment:
Maybe the error should be dependent on what we got back?

Reply:
That is the only possible cause today. If `Cursors_Purge` ever has multiple possible errors, we should use a `QueryError` and use it instead for the reply